### PR TITLE
50085_create_request_to_buy_a_new_book

### PR DESCRIPTION
**Pull request by zenith19**

### CODE REVIEW

File: app/helpers/request_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module RequestHelper
+end
```
Comment (MuhammadTamzid) [inline app/helpers/request_helper.rb]: remove this file

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -5,7 +5,15 @@
     <nav>
       <ul class="nav navbar-nav navbar-right">
         <li><%= link_to t(:home), root_path %></li>
+        <li class="dropdown">
+          <a href="#" class="dropdown-toggle" data-toggle="dropdown">Request</a>
+          <ul class="dropdown-menu">
+            <li><%= link_to t(:add_request), new_request_path %></li>
+            <li class="divider"></li>
+            <li><%= link_to t(:view_all_request), requests_path %></li>
```
Comment (MuhammadTamzid) [inline app/views/layouts/_header.html.erb]: don't need in this pull

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -5,7 +5,15 @@
     <nav>
       <ul class="nav navbar-nav navbar-right">
         <li><%= link_to t(:home), root_path %></li>
+        <li class="dropdown">
+          <a href="#" class="dropdown-toggle" data-toggle="dropdown">Request</a>
```
Comment (shakil-fg) [inline app/views/layouts/_header.html.erb]: i18n

File: config/routes.rb [hunk 4]
```diff
@@ -4,5 +4,6 @@
     get "home" => "static_pages#home"
     get "about" => "static_pages#about"
     get "contact" => "static_pages#contact"
+    resources :requests
```
Comment (MuhammadTamzid) [inline config/routes.rb]: only create for nthis pull

File: Gemfile [hunk 5]
```diff
@@ -16,6 +16,7 @@ gem "coffee-rails",            "4.1.0"
 gem "jquery-rails",            "4.0.3"
 gem "turbolinks",              "2.3.0"
 gem "jbuilder",                "2.2.3"
+gem "cancan"
```
Comment (sujoydatta) [inline Gemfile]: https://github.com/CanCanCommunity/cancancan should use this

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,6 +1,10 @@
 class ApplicationController < ActionController::Base
   before_action :set_locale
   protect_from_forgery with: :exception
+  rescue_from CanCan::AccessDenied do |exception|
+    flash[:error] = "Access denied."
```
Comment (sujoydatta) [inline app/controllers/application_controller.rb]: i18n

File: Gemfile (hunk 5 above)
Comment (MuhammadTamzid) [inline Gemfile]: use cancancan

**Comment by sujoydatta** [comment]
ok

File: app/controllers/requests_controller.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+class RequestsController < ApplicationController
+  load_and_authorize_resource
+
+  def new
+    @request = Request.new
```
Comment (dieunb) [inline app/controllers/requests_controller.rb]: no need, remove it. Do you know why? :D

File: app/controllers/requests_controller.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def create
+    @request = current_user.requests.new request_params
```
Comment (dieunb) [inline app/controllers/requests_controller.rb]: Remove it

File: app/models/request.rb [hunk 9]
```diff
@@ -1,3 +1,7 @@
 class Request < ActiveRecord::Base
   belongs_to :user
+  validates_presence_of :title
```
Comment (dieunb) [inline app/models/request.rb]: add white line above this, should split it to block

File: config/locales/vn.yml [hunk 10]
```diff
@@ -1,8 +1,16 @@
...
+  about: Trong khoảng
+  add_request: Thêm một yêu cầu
+  contact: Liên Hệ
+  dev_by: Được phát triển bởi
+  home: Nhà
+  request: yêu cầu
+  requests:
+    new:
+      flash_request_success: Yêu cầu tạo thành công
+      sent_request: Gửi yêu cầu
```
Comment (dieunb) [inline config/locales/vn.yml]: `Đã gửi yêu cầu`

File: app/controllers/requests_controller.rb (hunk 8 above)
Comment (MuhammadTamzid) [inline app/controllers/requests_controller.rb]: @request.user = current_user

**Pull request merged**